checks/microbenchmarks/gpu/amd_gpu/amd_gpu.py

```diff
@@ -51,29 +53,58 @@ def prepare_build(self):
 
     @run_before('run')
     def set_executable(self):
-        self.executable = f'{self.build_system.builddir}/radix-sort'
+        self.executable = f'{self.build_system.builddir}/{self.algo}'
         self.executable_opts = [self._executable_opts]
 
     @run_before('sanity')
     def set_sanity(self):
-        regex = r'radix sort time for.*key-value pairs:.*s, bandwidth:.*MiB/s'
+        if self.algo == 'radix-sort':
+            regex = r'radix sort time for.*key-value pairs:.*s, bandwidth:.*MiB/s'
+        elif self.algo == 'scan':
+            regex = r'exclusive scan time for.*values:.*s, bandwidth:.*MiB/s'
+        elif self.algo == 'reduce':
+            regex = r'reduction time for.*values:.*s, bandwidth:.*MiB/s'
+        else:
+            raise ValueError(f'Unknown algorithm: {self.algo}')
```

Comment:
Similar to the other comment, this could use `match-case` or a map (I prefer the latter).